### PR TITLE
Support Tier0 logical router as a resource

### PR DESCRIPTION
**Pull request by asarfaty**

### CODE REVIEW

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+			"high_availability_mode": &schema.Schema{
+				Type:         schema.TypeString,
+				Description:  "High availability mode",
+				Default:      "ACTIVE_STANDBY",
+				Optional:     true,
+				ForceNew:     true, // Cannot change the HA mode of a router
+				ValidateFunc: validation.StringInSlice(highAvailabilityValues, false),
+			},
+			"firewall_sections": getResourceReferencesSchema(false, true, []string{}, "List of Firewall sections related to the logical router"),
+			"edge_cluster_id": &schema.Schema{
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: Maybe this one is effectively ForceNew (I think you can't change edge cluster if its connected up)

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	d.SetId(logicalRouter.Id)
+
+	return resourceNsxtLogicalTier0RouterRead(d, m)
+}
+
+func resourceNsxtLogicalTier0RouterRead(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(*api.APIClient)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining logical Tier-1 router id")
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: Tier-0

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	}
+	if err != nil {
+		return fmt.Errorf("Error during LogicalTier0Router read: %v", err)
+	}
+
+	d.Set("revision", logicalRouter.Revision)
+	d.Set("description", logicalRouter.Description)
+	d.Set("display_name", logicalRouter.DisplayName)
+	setTagsInSchema(d, logicalRouter.Tags)
+	d.Set("advanced_config", logicalRouter.AdvancedConfig)
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: This one is not defined yet

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	d.Set("edge_cluster_id", logicalRouter.EdgeClusterId)
+	d.Set("high_availability_mode", logicalRouter.HighAvailabilityMode)
+	err = setResourceReferencesInSchema(d, logicalRouter.FirewallSections, "firewall_sections")
+	if err != nil {
+		return fmt.Errorf("Error during LogicalTier0Router firewall sections set in schema: %v", err)
+	}
+	d.Set("preferred_edge_cluster_member_index", logicalRouter.PreferredEdgeClusterMemberIndex)
+	d.Set("router_type", logicalRouter.RouterType)
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: is this needed?

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	d.Set("display_name", logicalRouter.DisplayName)
+	setTagsInSchema(d, logicalRouter.Tags)
+	d.Set("advanced_config", logicalRouter.AdvancedConfig)
+	d.Set("edge_cluster_id", logicalRouter.EdgeClusterId)
+	d.Set("high_availability_mode", logicalRouter.HighAvailabilityMode)
+	err = setResourceReferencesInSchema(d, logicalRouter.FirewallSections, "firewall_sections")
+	if err != nil {
+		return fmt.Errorf("Error during LogicalTier0Router firewall sections set in schema: %v", err)
+	}
+	d.Set("preferred_edge_cluster_member_index", logicalRouter.PreferredEdgeClusterMemberIndex)
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: shouldn't we expose this in schema?

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+	}
+
+	return resourceNsxtLogicalTier0RouterRead(d, m)
+}
+
+func resourceNsxtLogicalTier0RouterDelete(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(*api.APIClient)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining logical tier1 router id")
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: tier0

File: nsxt/resource_nsxt_logical_tier0_router.go [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	d.Set("router_type", logicalRouter.RouterType)
+
+	return nil
+}
+
+func resourceNsxtLogicalTier0RouterUpdate(d *schema.ResourceData, m interface{}) error {
+	nsxClient := m.(*api.APIClient)
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining logical tier1 router id")
```
Comment (annakhm) [inline nsxt/resource_nsxt_logical_tier0_router.go]: tier0

File: nsxt/resource_nsxt_logical_tier0_router.go (hunk 3 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_tier0_router.go]: Removed

File: nsxt/resource_nsxt_logical_tier0_router.go (hunk 6 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_tier0_router.go]: Fixed

File: nsxt/resource_nsxt_logical_tier0_router.go (hunk 7 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_tier0_router.go]: Fixed

File: nsxt/resource_nsxt_logical_tier0_router.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_tier0_router.go]: Done

File: nsxt/resource_nsxt_logical_tier0_router.go (hunk 2 above)
Comment (asarfaty) [reply nsxt/resource_nsxt_logical_tier0_router.go]: Fixed

**Pull request merged**